samples/snippets/fsharp/VS_Snippets_CLR_System/system.aggregateexception.class/fs/exception1.fs

```diff
@@ -0,0 +1,56 @@
+﻿// <Snippet1>
+open System
+open System.IO
+open System.Threading.Tasks
+
+let getAllFiles str =
+    // Should throw an UnauthorizedAccessException exception.
+    System.IO.Directory.GetFiles(str, "*.txt", System.IO.SearchOption.AllDirectories)
+
+// Get a folder path whose directories should throw an UnauthorizedAccessException.
+let path =
+    Environment.SpecialFolder.UserProfile
+    |> Environment.GetFolderPath
+    |> Directory.GetParent
+    |> fun x -> x.FullName
+
+// Use this line to throw an exception that is not handled.
+// let task1 = Task<string []>.Factory.StartNew(fun () -> raise (IndexOutOfRangeException()) )
+let task1 = Task.Factory.StartNew(fun () -> getAllFiles (path))
+
+try
```

Comment:
Put this kind of large code snippet involving a try/with in a function of its own